src/clj/fluree/db/json_ld/iri.cljc

```diff
@@ -14,7 +14,10 @@
 
 (defrecord IRI [^String s]
   Object
-  (toString [_] s))
+  (toString [_] s)
+  Comparable
+  (compareTo [_ o]
```

Comment:
Should we add an `IRI` type hint to `o` to make sure we only compare iris to each other?

Reply:
I thought about it, but after the javadoc for `Comparable` I saw that it defines the method in terms of T:
`Interface Comparable<T>`
[compareTo](https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/Comparable.html#compareTo(T))([T](https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/Comparable.html) o)

So I figured that I could assume that callers would do it correctly and that if it blows up when called incorrectly it should be fine. Also, when I added the type hint to the signature it throws a syntax error:
```
Syntax error (IllegalArgumentException) compiling deftype* at (src/clj/fluree/db/json_ld/iri.cljc:15:1).
Can't find matching method: compareTo, leave off hints for auto match.
```
So I just thought to leave it off. I can add a hint to the `(str o)` call, though.